File: src/utils/content.js
// DEPRECATED, use selectors instead

// import { rgb_to_hsv } from 'colorsys'
import { API_URL, THUMBS_URL } from '../utils/config'

export function getContentFromRecommendation(recommendation) {
  // check and unpack
  if (!recommendation) {
    return
  }
  const { mediation, recommendationOffers } = recommendation
  // choose one of the associated offer
  // for now we just pick randomly one of them
  // and this is actually what we want for the case where we have an event
  // proposed by several ticketers
  const chosenOffer =
    recommendationOffers &&
    recommendationOffers[
      Math.floor(Math.random() * recommendationOffers.length)
    ]
  // check
  if (!chosenOffer && !mediation) {
    return
  }
  // source
  let source
  let sourceCollectionName
  let venue
  if (!mediation && chosenOffer) {
    if (chosenOffer.eventOccurence) {
      source = chosenOffer.eventOccurence
      sourceCollectionName = 'eventOccurences'
    } else {
      source = chosenOffer.thing
      sourceCollectionName = 'things'
    }
  } else {
    if (mediation.event) {
      source = mediation.event
      sourceCollectionName = 'events'
      venue = chosenOffer.eventOccurence.venue
    } else {
      source = mediation.thing
      sourceCollectionName = 'things'
    }
  }
  // venue
  venue =
    venue || (chosenOffer && chosenOffer.venue) || (source && source.venue)
  // color
  let backgroundColor
  let thumbUrl
  // TODO: colorsys.rgb_to_hsv({ r: 255, g: 255, b: 255 }) ...
  if (mediation && mediation.thumbCount > 0) {
    thumbUrl = `${THUMBS_URL}/mediations/${mediation.id}`
    backgroundColor = mediation.firstThumbDominantColor
  } else if (source && source.thumbCount > 0) {
    thumbUrl = `${THUMBS_URL}/${sourceCollectionName}/${source.id}`
    backgroundColor = source.firstThumbDominantColor
  } else {
    backgroundColor = [0, 0, 0]
  }
  return Object.assign(
    {
      backgroundColor,
      chosenOffer,
      source,
      thumbUrl,
      venue,
    },
    recommendation
  )
}

File: utils/content.py
""" content """
from random import randint

def get_mediation (recommendation):
    return recommendation.get('mediation')

def get_stock (stock_id, recommendation):
    return [o for o in recommendation.get('recommendationStocks', []) if o['id'] == stock_id][0]

def get_source(mediation, stock):
    return stock.get('eventOccurrence', {"event": None}).get('event') or\
           stock.get('thing') or\
           mediation.get('event') or\
           mediation.get('thing')

def get_thumb_url(mediation, source, stock):
    source_plural_model_name = (stock.get('eventOccurrence') and 'events') or\
        (stock.get('thing') and 'things') or\
        (mediation.get('event') and 'events') or\
        (mediation.get('thing') and 'things') or ''
    if mediation and mediation.get('thumbCount') > 0:
        return '/mediations/'+ mediation['id']
    elif source and source.get('thumbCount') > 0:
        return source_plural_model_name + '/' + source['id']

def get_venue(stock, source):
    return stock.get('eventOccurrence', {'venue': None}).get('venue') or\
    stock.get('venue') or\
    source.get('eventOccurrence', {'venue': None}).get('venue') or\
    source.get('venue')

def get_content(recommendation):
    content = {}
    mediation = recommendation.get('mediation')
    recommendation_stocks = recommendation['recommendationStocks']
    stock_index = randint(0, len(recommendation_stocks) - 1)
    stock_id = recommendation_stocks[stock_index]['id']
    mediation = get_mediation(recommendation)
    stock = get_stock(stock_id, recommendation)
    source = get_source(mediation, stock)
    venue = get_venue(stock, source)
    content['thumbUrl'] = get_thumb_url(mediation, source, stock)
    content.update({
        "stock": stock,
        "mediation": mediation,
        "source": source,
        "venue": venue
    })
    return content
